.github/workflows/CI.yml

```diff
@@ -3,39 +3,58 @@ name: CI
 on:
   push:
     branches:
-      - main
-      - master
+      - "main"
+    paths:
+      - '.github/workflows/CI.yml'
+      - 'ext/**'
+      - 'src/**'
+      - 'test/**'
+      - 'Project.toml'
   pull_request:
+    paths:
+      - '.github/workflows/CI.yml'
+      - 'ext/**'
+      - 'src/**'
+      - 'test/**'
+      - 'Project.toml'
+  release:
   workflow_dispatch:
 
+concurrency:
+  # Skip intermediate builds: always.
+  # Cancel intermediate builds: always.
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
 jobs:
   test:
     name: Julia ${{ matrix.version }} - ${{ matrix.os }} - ${{ matrix.arch }} - ${{ github.event_name }}
     runs-on: ${{ matrix.os }}
+    timeout-minutes: 30
     strategy:
       fail-fast: false
       matrix:
         version:
-          - '1.10'  # Latest release
+          - "min"
         os:
           - ubuntu-latest
           - macOS-latest
           - windows-latest
         arch:
-          - x64
-
+          - "default"
     steps:
-      - uses: actions/checkout@v4
-
-      - uses: julia-actions/setup-julia@v1
+      - uses: actions/checkout@v5
+      - uses: julia-actions/setup-julia@v2
         with:
           version: ${{ matrix.version }}
           arch: ${{ matrix.arch }}
-
-      - uses: julia-actions/cache@v1
-
-      - uses: julia-actions/julia-buildpkg@v1
-
+      - uses: julia-actions/cache@v2
       - uses: julia-actions/julia-runtest@v1
         with:
           coverage: true
+      - uses: julia-actions/julia-processcoverage@v1
+      - uses: codecov/codecov-action@v5
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
Someone needs to actually create this token at https://app.codecov.io/account/gh/numericalearth/org-upload-token/ and add it as as an [org-wide actions secret](https://github.com/organizations/NumericalEarth/settings/secrets/actions) called `CODECOV_TOKEN`

Reply:
Tokenless uploads are a mess in codecov: https://github.com/codecov/codecov-action/issues/1293